esp-hal/src/uart.rs

```diff
@@ -1221,7 +1187,7 @@ where
     }
 
     /// Flush the transmit buffer of the UART
-    pub fn flush(&mut self) {
+    pub fn flush(&mut self) -> Result<(), Error> {
```

Comment:
My problem with this, apart from being hidden behind a PR that is supposed to just rename/unstabilize things, is that the hardware does not have any TX errors, and Error contains RX errors, and handling those aren't free.

Reply:
I just saw that `flush_async` was fallible and assumed it was an oversight, but I guess it was actually the other way around and async flush should also be infallible?

Reply:
I've backed these changes out of this PR.